OrbitCore/CrashHandler.cpp

```diff
@@ -10,89 +10,39 @@
 
 //-----------------------------------------------------------------------------
-void SendDumpInternal(const std::wstring& a_Dir, const std::wstring& a_Id);
-google_breakpad::ExceptionHandler* GHandler;
-
-//-----------------------------------------------------------------------------
-bool DmpFilter(void* context, EXCEPTION_POINTERS* exinfo,
-               MDRawAssertionInfo* assertion) {
-  return true;
-}
-
-//-----------------------------------------------------------------------------
-bool DmpCallback(const wchar_t* dump_path, const wchar_t* minidump_id,
-                 void* context, EXCEPTION_POINTERS* exinfo,
-                 MDRawAssertionInfo* assertion, bool succeeded) {
-  PRINT_FUNC;
-
-  std::string dir = Path::GetDumpPath();
-  std::string msg = "A crash dump was generated in " + dir;
-  MessageBox(NULL,
-             succeeded ? s2ws(msg).c_str() : L"Failed to generate crash dump",
-             L"Orbit Crash Handler: ", MB_ICONEXCLAMATION | MB_OK);
-
-  return false;
-}
-
-//-----------------------------------------------------------------------------
-bool OnDemandDmpCallback(const wchar_t* dump_path, const wchar_t* minidump_id,
-                         void* context, EXCEPTION_POINTERS* exinfo,
-                         MDRawAssertionInfo* assertion, bool succeeded) {
-  SendDumpInternal(dump_path, minidump_id);
-  return false;
-}
+crashpad::CrashpadClient* GCrashpadClient;
 
 //-----------------------------------------------------------------------------
 CrashHandler::CrashHandler() {
-  // Creates a new ExceptionHandler instance to handle writing minidumps.
-  // Before writing a minidump, the optional filter callback will be called.
-  // Its return value determines whether or not Breakpad should write a
-  // minidump.  Minidump files will be written to dump_path, and the optional
-  // callback is called after writing the dump file, as described above.
-  // handler_types specifies the types of handlers that should be installed.
-
-  assert(GHandler == nullptr);
-
-  /*GHandler = new google_breakpad::ExceptionHandler(
-      Path::GetDumpPath().c_str(),
-      DmpFilter,
-      DmpCallback,
-      0,
-      google_breakpad::ExceptionHandler::HANDLER_ALL,
-      MiniDumpNormal,
-      L"",
-      0 );*/
+  // Creates a new CrashpadClient instance that directs crashes to crashpad
+  // handler. Minidump files will be written to dump_path.
+
+  const std::wstring kHandlerPath(L"..\\bin\\crashpad_handler.exe");
+  base::FilePath dump_path(s2ws(Path::GetDumpPath().c_str()));
+  base::FilePath handler_path(kHandlerPath);
+
+  std::map<std::string, std::string> annotations;
+  std::vector<std::string> arguments;
+
+  GCrashpadClient = new crashpad::CrashpadClient();
+  GCrashpadClient->StartHandler(handler_path, /*database=*/dump_path,
+                                /*metrics_dir=*/dump_path, /*url=*/"",
+                                annotations, arguments, /*restartable=*/true,
```

Comment:
I think you can replace `annotations` and `arguments` just by `{}`.

For explanation: What you wrote is not wrong at all I just makes everybody else thinking why you
introduced a named variable here when you're only intention was to pass an empty map or vector along.